docs/compilers/CSharp/Compiler Breaking Changes - DotNet 10.md

```diff
@@ -89,3 +89,42 @@ class C
     }
 }
 ```
+
+## `UnscopedRefAttribute` cannot be used with old ref safety rules
```

Comment:
What happens if we use `[InlineArray]` in an older language version? I would expect that we have symetry between that behavior and this behavior. Basically compiler should have a uniform policy of how we approach seeing attributes in older language versions (unless there are specific reasons to deviate). Pretty our policy is to ignore but want to double check. 

Can we also update the general compiler documentation / feature check list to call out what behavior we want for attributes in this context?

Reply:
Ultimately it seems to better serve readers and writers of code to disallow use of `[UnscopedRef]` in cases where it is not having any effect. We are already doing this in several scenarios with latest LangVersion/TFM, where applying the attribute isn't defined to do something meaningful.

Reply:
The other part I'm interested in though is where do we document how attributes related to language features should be treated? This is a topic that's come up several times in the last few releases. Want us to find a place to codify what the default approach should be for such attributes. It also seems like an item that should be added to the test plan.

Reply:
Opened https://github.com/dotnet/roslyn/pull/76112.

Wanted to use `CallerArgumentExpressionAttribute` as an example, but turns out we don't ignore it in old LangVersions (probably not that big of an issue for that particular attribute).